src/ray/core_worker/reference_count.h

```diff
@@ -849,7 +849,9 @@ class ReferenceCounter : public ReferenceCounterInterface,
 
   /// Delete the object primary copy, if any. Also unsets the raylet address
   /// that the object was pinned at, if the address was set.
-  void DeleteObjectPrimaryCopy(ReferenceTable::iterator it);
+  void ResetObjectPrimaryCopy(ReferenceTable::iterator it);
+
+  void OnObjectOutOfScopeOrFreed(ReferenceTable::iterator it);
```

Comment:
Do we really need this rename? The intent behind the method doesn't really change

Reply:
I think the new name make it more clear when this method should be called. The bug is that we are calling it when it shouldn't be called.